pkgs/build-support/composer/fetchcomposer.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, composer }:
+
+{ name ? "composer-deps", sha256, ... } @ args:
+stdenv.mkDerivation ({
+  name = "${name}-vendor";
+  nativeBuildInputs = [ composer ];
+
+  phases = "unpackPhase patchPhase installPhase";
+
+  installPhase = ''
+    if [[ ! -f composer.lock ]]; then
+        echo
+        echo "ERROR: The composer.lock file doesn't exist"
+        echo
+        echo "composer.lock is needed to make sure that composerSha256 doesn't change"
+        echo "when the registry is updated."
+        echo
+
+        exit 1
+    fi
+
+    # Cannot use `COMPOSER_VENDOR_DIR` to install directly to `$out`, some packages use Composer
+    # events (`post-update`, `post-autoload-dump`, etc.) to require files relative to `vendor-dir`
+    # (which they request from `getComposer()->getConfig()->get(..)`) - unfortunately, this appears
+    # to return the default vendor path, not `COMPOSER_VENDOR_DIR`, so the file that they are
+    # looking for isn't found.
+    composer install -o
+    mv vendor/ $out/
+  '';
+
+  outputHashAlgo = "sha256";
+  outputHashMode = "recursive";
+  outputHash = sha256;
```

Comment:
It would be nice if this was not limited to `sha256` and supported SRI hashes like https://github.com/NixOS/nixpkgs/commit/267c8d6b2fea05bc811c8e2c2f4529b1436eeb9a (they are going to be more prominent with the advent of flakes).

Reply:
I've modified the derivations to allow the `outputHashAlgo` attribute to be provided too.